portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,14 +33,97 @@
-    response.setContentType("application/json;");
-    // response.getWriter().println("Hello Chisom!");
-    String jsonComments = "{\"Femi\": \"Nice page Chisom\", \"Dera\": \"Try to make the web page reponsive\", \"Kamsi\": \"Kudos\"}";
-    response.getWriter().println(jsonComments);
-  }
+   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      Query query = new Query("UserData");
+      ArrayList<UserData> data = new ArrayList<UserData>();	
+      PreparedQuery results = datastore.prepare(query);
+
+      Date currentDate = new Date();  
+    
+      for (Entity entity : results.asIterable()) { 
+        long id = entity.getKey().getId();
+        String name = (String) entity.getProperty("name");
+        String email = (String) entity.getProperty("email");
+        String comment = (String) entity.getProperty("comment");
+        Date date = (Date) entity.getProperty("date");
+        UserData newData = new UserData(id, name, email, comment, date);
+        data.add(newData);
+        }
+
+        response.setContentType("application/json;");
+
+        Gson gson = new Gson();
+        int numComments = getNumComments(request, "numValue", 5);
+        
+
+        if (numComments > data.size()){ 
+            response.getWriter().println(gson.toJson(data.subList(0, data.size())));
+        }
+        else{
+            response.getWriter().println(gson.toJson(data.subList(0, numComments )));
+        }
+    }
+
+    public static class UserData{
+        long id;
+        String name;
+        String email;
+        String comment;
+        Date date;
+
+        UserData(long id, String name, String email, String comment, Date date){
+            this.id = id;
+            this.name = name;
+            this.email = email;
+            this.comment = comment;
+            this.date = date;
+        }
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Date date = new Date();  
+        String name = getParameter(request, "name", "");
+        String email = getParameter(request, "email", "");
+        String comment = getParameter(request, "comment", "");
 
+        if(name.trim().isEmpty()){ 
+            throw new IllegalArgumentException("Name not entered.");}
 
+        if (comment.trim().isEmpty()){ 
+            throw new IllegalArgumentException("Comment field empty.");}
+
+        Entity taskEntity = new Entity("UserData");
+    	taskEntity.setProperty("name", name);
+   	 	taskEntity.setProperty("email", email);
+        taskEntity.setProperty("comment", comment);
+   	 	taskEntity.setProperty("date", date);
+        datastore.put(taskEntity);
+
+        // Redirect back to the HTML page.
+    	response.sendRedirect("/index.html");
+    }
+
+    private int getNumComments(HttpServletRequest request,  String name, int defaultValue) {
```

Comment:
if I call:

getNumComments(request, "definitelyNotComments", ...)

is this retrieving the number of comments? Or is it retrieving the number of something else? If I can use this function to get things that are not the number of comments, should it be called getNumComments()? What might be a better name, that tells us what this function is actually doing?

Reply:
I renamed it to "getNumCommentsUserSelects" ? Its a bit long, but i believe it explains what the method is doing. The function returns an integer which represents a value the user selected on the HTML menu. The method's parameters are a servlet request, a string( the selection menu Id ), and a default value. If the user made no selection, the method returns a default number.